src/entities/createPool/weighted/createPoolWeighted.ts

```diff
@@ -0,0 +1,66 @@
+import { Address, encodeFunctionData, parseEther, zeroAddress } from 'viem';
+import {
+    CreatePoolBase,
+    CreatePoolBuildCallOutput,
+    CreateWeightedPoolArgs,
+    WeightedCreatedPoolInput,
+} from '../types';
+import { getRandomBytes32 } from '../../utils/getRandomBytes32';
+import { weightedFactoryV4Abi } from '../../../abi/weightedFactoryV4';
+
+export class CreatePoolWeighted implements CreatePoolBase {
+    buildCall(input: WeightedCreatedPoolInput): CreatePoolBuildCallOutput {
+        const args = this.parseCreateFunctionArgs(input);
+        const encodedCall = encodeFunctionData({
+            abi: weightedFactoryV4Abi,
+            functionName: 'create',
+            args,
+        });
+        return { call: encodedCall };
+    }
+
+    private parseCreateFunctionArgs(
+        input: WeightedCreatedPoolInput,
+    ): CreateWeightedPoolArgs {
+        const sortedTokenParams = input.tokens.map((tokenAddress) => {
+            let weight: number = 1 / input.tokens.length;
```

Comment:
What happens for pools with 3 tokens? Sum won't be exactly 1, right? Is that a problem?

Reply:
Connecting to the other comment, if you make `weights` required, than you won't have to worry about creating a test case for this.